jax/_src/numpy/fft.py

```diff
@@ -252,30 +258,35 @@ def fftfreq(n: int, d: ArrayLike = 1.0, *, dtype=None) -> Array:
           "The d argument of jax.numpy.fft.fftfreq only takes a single value. "
           "Got d = %s." % list(d))
 
-  k = jnp.zeros(n, dtype=dtype)
+  k = jnp.zeros(n, dtype=dtype, device=device)
   if n % 2 == 0:
     # k[0: n // 2 - 1] = jnp.arange(0, n // 2 - 1)
-    k = k.at[0: n // 2].set( jnp.arange(0, n // 2, dtype=dtype))
+    k = k.at[0: n // 2].set( jnp.arange(0, n // 2, dtype=dtype, device=device))
 
     # k[n // 2:] = jnp.arange(-n // 2, -1)
-    k = k.at[n // 2:].set( jnp.arange(-n // 2, 0, dtype=dtype))
+    k = k.at[n // 2:].set( jnp.arange(-n // 2, 0, dtype=dtype, device=device))
 
   else:
     # k[0: (n - 1) // 2] = jnp.arange(0, (n - 1) // 2)
-    k = k.at[0: (n - 1) // 2 + 1].set(jnp.arange(0, (n - 1) // 2 + 1, dtype=dtype))
+    k = k.at[0: (n - 1) // 2 + 1].set(jnp.arange(0, (n - 1) // 2 + 1, dtype=dtype, device=device))
 
     # k[(n - 1) // 2 + 1:] = jnp.arange(-(n - 1) // 2, -1)
-    k = k.at[(n - 1) // 2 + 1:].set(jnp.arange(-(n - 1) // 2, 0, dtype=dtype))
+    k = k.at[(n - 1) // 2 + 1:].set(jnp.arange(-(n - 1) // 2, 0, dtype=dtype, device=device))
 
-  return k / jnp.array(d * n, dtype=dtype)
+  return k / jnp.array(d * n, dtype=dtype, device=device)
```

Comment:
@jakevdp will this work for the length-n sharding ?

Reply:
Since this is divisible by `n` I think it will, but we're going to have to add some tests for these kinds of corner cases.